### PR TITLE
add admissionRequest in ctx

### LINKED ISSUE #2656: Consider putting the admission review on the webhook context
> The use case is that I need to parse again the AdmissionRequest from Request.Body to perform further checks in the validating/mutating phase.

Would it make more sense for the `AdmissionRequest` go object to be on the context - instead of having the raw request body and parsing it again yourself?

_Originally posted by @dprotaso in https://github.com/knative/pkg/issues/2583#issuecomment-1331507771_

### PR DESCRIPTION
**Pull request by rahulii**
# Changes

-  :broom: This PR associated the admissionv1.AdmissionRequest object the webhook received with the context.
-  adds a unit test

/kind <kind>

Fixes #2656 

**Release Note**

**Docs**

### CODE REVIEW

File: apis/contexts.go [hunk 1]
```diff
@@ -245,6 +246,7 @@ func IsDryRun(ctx context.Context) bool {
 // This is attached to contexts passed to webhook interfaces with
 // additional context from the HTTP request.
 type httpReq struct{}
+type AdmissionReq struct{}
```
Comment (dprotaso) [inline apis/contexts.go:249]: don't expore this
```suggestion
type admissionRequest struct{}
```

**Review by dprotaso — CHANGES_REQUESTED** [review]
We should remove the HTTP request from the context in favour of the AdmissionRequest

**Comment by rahulii** [comment]
> We should remove the HTTP request from the context in favour of the AdmissionRequest

https://github.com/knative/pkg/blob/main/webhook/conversion.go#L59
converion webhook also uses this function - we would not be able to remove it I guess unless we also support `Request *ConversionRequest`

**Comment by dprotaso** [comment]
> conversion webhook also uses this function - we would not be able to remove it I guess unless we also support Request *ConversionRequest

Yeah let's do that and add helpers for `ConversionReview`

**Pull request closed without merging**